backend/src/packages/users/user.repository.ts

```diff
@@ -23,40 +20,26 @@ class UserRepository implements Repository {
     return Promise.resolve(null);
   }
 
-  public async findAll(): Promise<UserEntity[]> {
-    const users = await this.userModel.query().execute();
-
-    const userPromises = users.map(async (user) => {
-      const userJoinedWithUserDetails = await this.getUserJoinWithUserDetails(
-        user.id,
-      );
-      return UserEntity.initialize(userJoinedWithUserDetails);
-    });
-    return await Promise.all(userPromises);
-  }
-
-  private flattenUserJoinWithUserDetails(
+  private mapUserWithUserDetailsJoin(
```

Comment:
Remove this helper and map manually, like:
```
 return users.map((user) => {
      return UserEntity.initialize({
        id: user.id,
        email: user.email,
        passwordHash: user.passwordHash,
        passwordSalt: user.passwordSalt,
        firstName: user.userDetails.firstName,
        lastName: user.userDetails.lastName,
        // other properties
      });
    });
```

Reply:
Ok, but could you explain why not to use mapper?

Reply:
because we may have different fields for all the current methods, also we may map them differently in the future. So we can either create separate mappers for each method or just map them in the same function and we would use the second way